backend/gn_plugin_depobio/migrations/01eb67fd7851_add_depobio_plugin_module_r_permission.py

```diff
@@ -0,0 +1,72 @@
+"""Add depobio module permission
+
+Revision ID: 1e2926e0b603
+Revises: 78ba67f597ee
+Create Date: 2025-08-28 18:55:23.047899
+
+"""
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "01eb67fd7851"
+down_revision = None
+branch_labels = ("plugin_depobio",)
+depends_on = ("f051b88a57fd",)  # geonature: creation of gn_permissions.t_permissions_available
+
+module_code = "PLUGIN_DEPOBIO"
+object_code = "ALL"
+action_code = "R"
+
+
+def upgrade():
+    op.execute(
+        f"""
+        INSERT INTO
+            gn_permissions.t_permissions_available (
+                id_module,
+                id_object,
+                id_action,
+                label,
+                scope_filter
+            )
+        SELECT
+            m.id_module,
+            o.id_object,
+            a.id_action,
+            v.label,
+            v.scope_filter
+        FROM
+            (
+                VALUES
+                    ('{module_code}', '{object_code}', '{action_code}', False, 'Accéder au module')
+            ) AS v (module_code, object_code, action_code, scope_filter, label)
+        JOIN
+            gn_commons.t_modules m ON m.module_code = v.module_code
+        JOIN
+            gn_permissions.t_objects o ON o.code_object = v.object_code
+        JOIN
+            gn_permissions.bib_actions a ON a.code_action = v.action_code
+        """
+    )
+
+
+def downgrade():
+    op.execute(
```

Comment:
```suggestion
def downgrade():
    op.execute(
        f"""
        DELETE FROM gn_permissions.t_permissions
        WHERE
            id_module = (
                SELECT id_module
                FROM gn_commons.t_modules
                WHERE module_code = '{module_code}')
            AND id_object = (
                SELECT id_object
                FROM gn_permissions.t_objects
                WHERE code_object = '{object_code}')
            AND id_action = (
                SELECT id_action
                FROM gn_permissions.bib_actions
                WHERE code_action = '{action_code}');
        """
    )
    op.execute(
```

Je n'ai pas vérifié à nouveau, mais je crois que la gestion du ON DELETE de `t_permissions_available` vers `t_permissions` interdit le DELETE si entrées correspondantes dans la seconde table.

Si tel est le cas, commencer par supprimer les éventuelles permissions attribuées - entrées de la table `gn_permissions.t_permissions` - pour être sûr de pouvoir ensuite supprimer la permission disponible, i.e. l'entrée dans `gn_permissions.t_permissions_available`.

Reply:
Ah oui, possible. Mais, si c'est le cas, on a pas forcément envie que le downgrade automatique soit possible. Pour moi on accepte le downgrade automatique s'il n'y a pas de perte de données (donc que si je rejoue la migration, j'arrive au même état qu'au début). J'ai fait de la même façon que ce qui existait déjà sur GN